### PR TITLE
SPCTR-3452 : Adding auto and equal width functionality in block's which present inside the container

### PR DESCRIPTION
**Pull request by rahulvarma722**
### Description
We are currently working on enhancing the container block by adding two functionalities: 'auto' and 'equal width.' Currently, the container block uses the 'flex' property, ensuring that every block inside it has equal width. We are now introducing an option for users to choose between 'equal width' and 'auto width.' If a user selects 'auto width,' the blocks within the container will adjust their width automatically based on the content within them.

### Checklist:
- [ ] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -223,6 +223,8 @@ Our external packages use [Rating Star Component](https://github.com/n49/react-s
 * Fix: Global - Resolved an issue in which the Spectra blocks were unable to be previewed in the widget editor customizer.
 * Fix: Buttons - Resolved an issue in which the margins added to the button group would get overridden by child button margins.
 * Fix: Table Of Contents - Resolved an issue wherein the Table Of Contents collapse feature would not work as intended.	
+* New: Container - Introducing Auto Width and Equal Width options in Container blocks.
```
[the remark this answers is not in the document]
Comment (bsf-zanev) [reply README.md:226]: Let's move this changelog entey right on top.

File: src/blocks/container/attributes.js [hunk 2]
```diff
@@ -1525,6 +1525,30 @@ const attributes = {
 		},
 	},
 	...inputBorderAttributes,
+	childrenWidthDesktop: {
+		type: 'string',
+		default: '',
```
Comment (bsf-zanev) [inline src/blocks/container/attributes.js:1530]: Is empty default required in JS Attributes?

**Review by bsf-zanev — COMMENTED** [review]
Added a few minor comments.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply README.md:226]: Done

File: src/blocks/container/utils.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    const varDeviceType = 'childrenWidth' + deviceType;
+    const deviceLabel = {
+        Desktop: __( 'Desktop', 'ultimate-addons-for-gutenberg' ),
+        Tablet: __( 'Tablet', 'ultimate-addons-for-gutenberg' ),
+        Mobile: __( 'Mobile', 'ultimate-addons-for-gutenberg' ),
+    };
+
+    return (
+        <UAGSelectControl
+            label={__( 'Children Width ', 'ultimate-addons-for-gutenberg' ) + deviceLabel[ deviceType ] }
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply src/blocks/container/utils.js:19]: We need suggestions from designer for label text so in future we will change this after previewing by pos.

File: src/blocks/container/attributes.js (hunk 2 above)
Comment (rahulvarma722) [reply src/blocks/container/attributes.js:1530]: Removed empty string

File: src/blocks/container/styling.js [hunk 4]
```diff
@@ -803,6 +806,40 @@ function styling( attributes, clientId, name, deviceType, gbsSelector = false )
...
 	}
 
+	const flexDirections = [ 'row-reverse', 'row' ];
+	const autoWidth = { 'width': 'auto' };
+	const SetWidth = { 'width': '100%' };
+
+	// Add auto width to the inner blocks in desktop.
+	if( directionDesktop ){
+		if( flexDirections.includes( directionDesktop ) && 'auto' === childrenWidthDesktop ) {
+			selectors[ '.wp-block-uagb-container > .block-editor-inner-blocks > .block-editor-block-list__layout > .wp-block' ] = autoWidth;
```
[the remark this answers is not in the document]
Comment (rahulvarma722) [reply src/blocks/container/styling.js:816]: Done

**Review by bsf-zanev — APPROVED** [review]

File: src/blocks/container/utils.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (bsf-zanev) [reply src/blocks/container/utils.js:19]: Keeping this comment open until the required updates are done to this control after the decision has been made.

CC - @rahulvarma722, @adi3890.

File: src/blocks/container/styling.js [hunk 5]
```diff
@@ -803,6 +806,42 @@ function styling( attributes, clientId, name, deviceType, gbsSelector = false )
 		};
 	}
 
+	const flexDirections = [ 'row-reverse', 'row' ];
+	const autoWidth = { 'width': 'auto' };
+	const SetWidth = { 'width': '100%' };
```
Comment (adi3890) [inline src/blocks/container/styling.js:811]: should we use setWidth
starting with small 's' as per our naming convention.
Please check.

**Review by adi3890 — APPROVED** [review]

**Review by adi3890 — APPROVED** [review]

**Pull request merged**